### PR TITLE
Clerk info requests

### PR DESCRIPTION
**Pull request by zonotope**
The changes here are mostly cleanup because I realized that `voter-registration-works` does most of what we need it to do already. I nuked some old code that would never actually be reached in practice like the `state-specific-register-voter-fn` (and got rid of the auto namespace resolving stuff to boot), and moved the election authority lookup stuff to its own namespace. I think there could be more done around that, but it's beyond the scope of this task.

There are still some open questions though. In order to support this change, we still need to set the new hampshire state level election authority to `:call-clerk`, but I can't seem to find where/how to do that since there don't look to be any reg methods in the authority spreadsheet. Also, I think we want to make our registration methods enums as well, but they're just keywords now. That is also outside of scope.

### CODE REVIEW

File: src/voter_registration_works/states.clj [hunk 1]
```diff
@@ -65,14 +43,6 @@
...
     (registration-methods->by-type reg-methods)))
 
-(defn state-specific-register-voter-fn
-  "Returns a state-specific `register-voter` fn if it exists, nil otherwise."
-  [state method]
-  (try (ns-resolve
-        (symbol (str "voter-registration-works.states." (name state)))
-        (symbol (str "voter-register-" method)))
-       (catch Exception e nil)))
-
```
Comment (cap10morgan) [inline src/voter_registration_works/states.clj]: I don't want to get rid of this or the supporting code that calls it. It's all there to support future state voter reg. API integrations. I had hoped we'd get to integrating those sooner than we have, but we still plan to do so.

Comment (zonotope) [reply src/voter_registration_works/states.clj]: ok, that makes sense. i'll add it back.

**Comment by zonotope** [comment]
@cap10morgan i added back the state specific voter reg lookup, but i think, whenever we do get around to integrating with state voter reg apis, we might want to consider switching to a map with the states as key namespaces, the method as the key name, and the specific reg function as value instead of auto resolving namespaces from constructed strings. i think something like:

```clojure
{:ny/by-mail (fn [voter] (do something)
 ...}
```

is more explicit so it's easier to track what's going on, and to trace the code paths that reference those functions using things like editor code searches and such. I think that's outside of the scope of this pr, so it's just a suggestion for the future.

**Comment by cap10morgan** [comment]
@zonotope Yeah, we may need to alter the election authority spreadsheet and data model slightly here. If we added an `other-voter-reg-methods` field, for example, we could add `:call-clerk` to NH but leave the others alone (and allow them to keep their current behavior of offering by-mail reg. for any forms in `reg-by-mail-forms` and OVR when `ovr-url` has a value.

What do you think?

**Pull request merged**